### PR TITLE
Create Plugin that allows for automate database exports to a specified location

### PR DESCRIPTION
**Pull request by GraphiteWingedBird**
There are a number of various issues/questions that would be well served if the application had a built in backup feature, [Guide users to regularly create database backups](https://github.com/PhilippC/keepass2android/issues/46) and [Database backup function](https://github.com/PhilippC/keepass2android/issues/176) as two prime examples.

If this can't be handled by a plugin, it would be nice to have in the core code. There is already an option to "Export Database" within the Database Settings menu, but it requires you to take multiple steps that shouldn't be necessary and could be remembered as preferred settings. Current workflow:

1. Open a database
2. Click Settings
3. Click Database Settings
4. Click Export Database...
5. Choose format (.kdbx, .xml, .csv)
6. Select a storage location/type
7. Type in a filename
8. Click Save/Finish the activity

Ideally, we could have triggers designed that 'automatically' create a backup when 'something' happens. E.g. whenever you open a specific database. The automatic portion would be to either set ahead of time "preferred settings" or have those populated during the first export. I.e. the application should remember what file format to export as, where to save the backup, and what the name of the backup should be (with placeholders and wild cards potentially). Depending on how it is implemented, you might set a maximum number of backups (e.g. 5) and then the application could delete the oldest backup when it creates the newest to keep that number.

KeePass itself supports triggers, but the most useful for a backup are probably when the database first opens, when it is updated and/or saved, when it is synchronized, and when it is closed. Depending on the implementation, users could select which options create a backup for a given database, or they could flexibly define their own triggers. Since the Export Database Settings are database specific, I imagine that these types of backup settings would be database specific and either set within the Database Settings themselves, or set at the Application Level where you have an option to define which .kdbx files should backup

### CODE REVIEW

**Comment by PhilippC** [comment]
why did you request to merge the old fdroid branch? was this intentional?

**Comment by GraphiteWingedBird** [comment]
TL;DR - Not intentional, no clue what the branches are. Just want something
on "Android" developed and that request is the culmination of an hour of
sifting through GitHub resources and then clicking "Submit"

Definitely not intentional - my goal was to make a draft Pull Request
because even after reading the documentation on what a pull request is, I
don't really understand what they are other than an ability to 'create your
own branch' that will, presumably, at some point be merged into the
base/tree.

Like I mentioned in a comment, there is a very good chance I'd be more
successful just figuring out how to create a plugin in Java rather than
creating a pull request. I do have a buddy that knows GitHub though so I
can ask him for guidance on the error of my ways.

- Kris

**Comment by PhilippC** [comment]
this should be a ticket, no PR
